### PR TITLE
Fix: Separate product reference and manufacturer

### LINKED ISSUE #569: Product reference only shown if manufacturer is set
If product does not have manufacturer set the product reference wont be shown. 

See here: https://github.com/PrestaShop/hummingbird/blob/develop/templates/catalog/_partials/product-details.tpl#L17

Also the block naming is kinda misleading as the block is named product_reference but it contains both manufacturer and reference information.

Also would be really nice to have specific class for each `detail` li element. Maybe the `detail__title` information could be used to generate class for the li (does smarty have a `slugify` formatter)? That would allow doing things like highlighting one specific detail with just css.

### PR DESCRIPTION
**Pull request by dennispw**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Separates product reference and manufacturer information to different blocks and if statements.
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #569

### CODE REVIEW

File: templates/catalog/_partials/product-details.tpl [hunk 1]
```diff
@@ -29,17 +29,19 @@
...
-                <div class="detail__left">
-                  <span class="detail__title">{l s='Reference' d='Shop.Theme.Catalog'}</span>
-                </div>
-                <div class="detail__right">
-                  <span>{$product.reference_to_display}</span>
-                </div>
-              </li>
-            {/if}
+        {block name='product_reference'}
+          {if isset($product.reference_to_display) && $product.reference_to_display neq ''}
```
Comment (SharakPL) [inline templates/catalog/_partials/product-details.tpl:36]:
```suggestion
          {if !empty($product.reference_to_display)}
```
Should be enough

**Review by SharakPL — APPROVED** [review]
LGTM

**Review by boherm — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @tswfi ,

Tested with all types of products. LGTM
Tested with/out supplier, with/out supplier ref, with/out combination ref.

https://github.com/PrestaShop/hummingbird/assets/16019289/031530bd-7de4-424b-9391-599227342f0d

Thanks!

**Comment by SharakPL** [comment]
Thanks @dennispw and @florine2623

**Pull request merged**